R/centrality.R

```diff
@@ -1487,25 +1487,20 @@ alpha.centrality.dense <- function(graph, nodes = V(graph), alpha = 1,
   exo <- rep(exo, length.out = vcount(graph))
   exo <- matrix(exo, ncol = 1)
 
-  if (is.null(weights) && "weight" %in% edge_attr_names(graph)) {
-    ## weights == NULL and there is a "weight" edge attribute
-    attr <- "weight"
-  } else if (is.null(weights)) {
-    ## weights == NULL, but there is no "weight" edge attribute
-    attr <- NULL
+  if (is.null(weights) && is_weighted(graph)) {
+    weights <- get_weights(graph)
   } else if (is.character(weights) && length(weights) == 1) {
```

Comment:
@szhorvat should the functions `alpha.centrality.dense()` and `alpha.centrality.sparse()` actually error when provided `weights` as a character?